cmd/miniooni/main.go

```diff
@@ -155,37 +166,29 @@ func (h *logHandler) HandleLog(e *log.Entry) (err error) {
 }
 
 func main() {
+	logger := &log.Logger{
+		Level:   log.InfoLevel,
+		Handler: &logHandler{Writer: os.Stderr},
+	}
+	log.Log = logger // from now on logs may be redirected
+	log.Info("Stared...")
 	getopt.Parse()
 	if len(getopt.Args()) != 1 {
 		log.Fatal("You must specify the name of the experiment to run")
 	}
 	extraOptions := mustMakeMap(globalOptions.extraOptions)
 	annotations := mustMakeMap(globalOptions.annotations)
 
-	logger := &log.Logger{
-		Level:   log.InfoLevel,
-		Handler: &logHandler{Writer: os.Stderr},
-	}
```

Comment:
> I would strongly recommend to keep the logger here...

I moved the `logger` because `log.Fatal` was not printed according to our structure! (Duality in displaying logs)

>  ...because you'd be printing before the program has been able to decide whether it has been called with the correct arguments,..

I wrote a simple text, because otherwise it would require more explanation to get started. Like: `web connectivity test started` or something like that. And that required adding more code to the *mini* program. Maybe I can handle this by adding a separate pkg*?

* To handle printing progress percentages(if possible) and more accurate descriptions. 

> (Also, I believe there are better ways to give a sense of progress.)

I have tried different ways to show its progress.
Showing progress on a single line was the worst. All of them use `\r`, which makes the simultaneous display of logs very complicated. (Even deletes the original logs in some cases.)

(Is there a way to get a percentage of progress from MK and other test methods?)

Reply:
> [...] because log.Fatal was not printed according to our structure!

Did you test that?

> I wrote a simple text, because otherwise it would require more explanation to get started.

I suggest to write `Running with input ${input}` in the loop way down for every non-empty-string input. That seems much more useful, as a progress estimator, than "starting" because it tells you what input is being processed.

Reply:
> Showing progress on a single line was the worst. All of them use \r, which makes the simultaneous display of logs very complicated. (Even deletes the original logs in some cases.)

Please, just one log line for event. If we ever want to be fancy, we do that in `ooni/probe-cli`, not here.

> (Is there a way to get a percentage of progress from MK and other test methods?)

Yes. You get the `inputs` list length and then compute the percentage as you go. Second thinking about my above suggestion, I guess we can print `[33%] running with input ${input}`.

Reply:
> Did you test that?

This line is displayed with the default structure:
https://github.com/ooni/probe-engine/blob/c8e30597c66f5d05cb79486450f3ecdf18c8adb0/cmd/miniooni/main.go#L177

> ..., than "starting" because it tells you what input is being processed.

Good suggestion. But in some (non-verbose) tests, the first message is delayed by more than 5 seconds! No problem if no message is displayed before that?
```
# ./miniooni -no templog.txt web_connectivity 
[      0.000138] <info> Stared...
[     16.723880] <info>  O_o  Oh! Something is wrong. Please try again with the -v option.
[     16.740180] <info> ...Finished with error! Duration : 16.739705905s
```
```
# ./miniooni -no templog.txt web_connectivity 
[      0.000145] <info> Stared...
[     41.552178] <info>  O_o  Oh! Something is wrong. Please try again with the -v option.
[     41.552219] <fatal> cannot lookup OONI backends: map[error:All available bouncers failed]
```

> Please, just one log line for event.

OK! :D 

> Second thinking about my above suggestion 

Yeah! :D,  I had read the messages from bottom to top and answered the same way. :))

Reply:
> This line is displayed with the default structure:

Yeah, it actually should just print without any specific structure.

> the first message is delayed by more than 5 seconds

Then I suggest to print the high level operation being performed (e.g. getting information on available OONI backends)